docs/primitives/virtualgrid.md

```diff
@@ -0,0 +1,72 @@
+# VirtualGrid Primitive
+
+`VirtualGrid` renders a dynamic slice of items based on the currently selected index and supports smooth vertical scrolling between rows.
+
+### Behavior
+
+- Renders a 2D grid of items using `itemsPerRow` and `numberOfRows`.
+- Uses `rowsBuffer` to pre-render additional rows above and below the visible window.
+- Only a subset of the total items is rendered, improving performance.
+- Handles directional navigation (`onUp`, `onDown`, `onLeft`, `onRight`) using built-in key handlers.
+- Automatically scrolls vertically when the user navigates to a new row.
+- Triggers `onEndReached` when the user approaches the end of the list. Allowing fetching additional items
+- Focus is updated and maintained internally, with optional control via `scrollToIndex`.
+- `selected` can be set to the index in the total array of items (which can be gotten from the cursor property)
+
+### Example Usage
+
+```tsx
+import { VirtualGrid } from './components/VirtualGrid';
+
+<VirtualGrid
+  id="BrowseGrid"
+  x={160}
+  y={24}
+  width={1620}
+  scroll="always"
+  autofocus
+  itemsPerRow={7}
+  numberOfRows={2}
+  rowsBuffer={2}
+  each={provider().pages()}
+  onEnter={onEnter}
+  onEndReached={onEndReached}
+  onSelectedChanged={updateContentBlock}
+  announce={`All Trending ${props.params.filter}`}
+>
+  {(item) => <Thumbnail item={item()} />}
+</VirtualGrid>;
+```
+
+### Props
+
+- **each** (`readonly T[] | undefined | null | false`): The full list of items to be rendered.
+- **itemsPerRow** (`number`): Number of items per row (required).
+- **numberOfRows** (`number`): Number of visible rows (default: `1`).
+- **rowsBuffer** (`number`): Number of rows to pre-render above and below the visible area (default: `2`).
+- **onEndReached** (`() => void`): Callback triggered when selection moves near the end of the list.
+- **children** (`(item: Accessor<T>, index: Accessor<number>) => JSX.Element`): Function that renders each item.
+- **selected** (`number`): Initial selected index.
+- **scrollToIndex** (`(index: number) => void`): Imperatively scrolls and focuses the item at the given index.
```

Comment:
What is this? `scrollToIndex` is not a part of props type, nor the prop is used anywhere.